<?xml version="1.0"?>
<launch>
    <param name="use_sim_time" value="true" />
    <arg name="dataset" value="iot/realsense_good-points"/>
    <arg name="visualization" value="true"/> <!-- This is for debug publisher-->
    <arg name="publish_refined_keyframe" value="true"/>
    <arg name="parent_frame" value="ni_slam_base"/>
    <arg name="child_frame" value="camera_link"/>
    <param name="use_sim_time" value="true" />
    <param name="/map/resolution" value="0.005" />
    <arg name="queue_size_cloud" value="3"/>
    <arg name="queue_size_imu" value="10"/>
    <arg name="queue_size_sync" value="50"/>
    <param name="/axonometric/image_height" value="240" />
    <param name="/axonometric/image_width" value="320" />
    <!-- this is for facing down camera -->
    <!-- <rosparam param="rotation/imu"> [0, 3.1415926, 1.5707963] </rosparam> -->

    <!-- facing right -->
    <rosparam param="rotation/imu"> [0, 1.5707963, 3.1415926] </rosparam>

    <node pkg="rosbag" type="play" name="play_bag_node" args=" --queue=1000 --clock $(find ni_slam)/bag/$(arg dataset).bag" output="screen"/> 

    <node pkg="ni_slam" type="ni_slam_node" name="ni_slam_node" output="screen">
        <remap from="/imu/data" to="/viconxbee_node/mocap/imu"/>
        <remap from="/imu/data" to="/imu/vicon/data"/>
        <!-- <remap from="/camera/rgb/points" to="/camera/depth_registered/points"/> -->
        <param name="saveposes/filename_prefix" value="$(find ni_slam)/bag/$(arg dataset)" />
        <!--comment above line if you don't want to save any files, 
            or any existing files would be replaced.-->

        <param name="topic/time_diff" value="0.03" />
        <param name="train/psr_bound" value="40.0" />
        <param name="frame_name/parent" value="$(arg parent_frame)" />
        <param name="frame_name/child" value="$(arg child_frame)" />
        <param name="debug/visualization" value="$(arg visualization)" />
        <param name="refined_frame/publish_flag" value="$(arg publish_refined_keyframe)" />
        <param name="queue_size/cloud" value="$(arg queue_size_cloud)" />
        <param name="queue_size/imu" value="$(arg queue_size_imu)" />
        <param name="queue_size/sync" value="$(arg queue_size_sync)" />
    </node>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_ni_slam" output="screen" ns="est" >
        <param name="/target_frame_name" value="$(arg parent_frame)" />
        <param name="/source_frame_name" value="$(arg child_frame)" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="hector_trajectory_server_real" output="screen" ns="real" >
        <param name="/target_frame_name" value="$(arg parent_frame)" />
        <param name="/source_frame_name" value="robot_base" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node> 

  <node pkg="tf" type="static_transform_publisher" name="ni_slam_base_to_vicon_tf" args="0 0 0 3.1415926 0  0  vicon ni_slam_base 50" />

</launch>
